### PR TITLE
Document custom image transformations

### PR DESCRIPTION
**Pull request by aniketmaurya**
## What does this PR do?

Document custom image transformations

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/reference/image_classification.rst [hunk 1]
```diff
@@ -55,6 +55,17 @@ Here's the full example:
     :language: python
     :lines: 14-
 
+Flash automatically takes care of Image transformations for you. But sometimes it is not enough and you want to customize the transformations.
+You can easily customize `Preprocessing <https://github.com/PyTorchLightning/lightning-flash/blob/master/flash/core/data/process.py>`_ method in the `data_module`.
+The base `Preprocess` class defines 7 methods for different stages in the data loading pipeline.
+To apply image augmentations you can directly use `default_transforms` defined in Flash Image module and then add your merge your custom defined image transformations with it.
+In this example, we will load the default transforms and merge it with custom `torchvision` transformations. We use `post_tensor_transform` since the transformations will be applied
+after image is converted into `torch.Tensor`.
```
Comment (ethanwharris) [inline docs/source/reference/image_classification.rst:63]:
```suggestion
In this example, we will load the default transforms and merge with custom `torchvision` transformations. We use the `post_tensor_transform` hook to apply the transformations
after the image is converted to a `torch.Tensor`.
```

File: docs/source/reference/image_classification.rst [hunk 2]
```diff
@@ -55,6 +55,17 @@ Here's the full example:
...
+
+.. literalinclude:: ../../../flash_examples/image_transforms.py
+    :language: python
```
Comment (ethanwharris) [inline docs/source/reference/image_classification.rst:66]: Let's use testcode here and have the example directly in the docs.

File: docs/source/reference/image_classification.rst [hunk 3]
```diff
@@ -55,6 +55,17 @@ Here's the full example:
     :language: python
     :lines: 14-
```
Comment (ethanwharris) [inline docs/source/reference/image_classification.rst:57]: I think we could have a new section for this. Don't mind what you call it, `Custom Transformations` or something like that?

**Review by ethanwharris — COMMENTED** [review]
Awesome work! Few small comments on the text. I think we should move the example to a testcode block, let me know if you need any help with that :smiley:

File: docs/source/reference/image_classification.rst (hunk 3 above)
Comment (aniketmaurya) [reply docs/source/reference/image_classification.rst:57]: Yes, `Custom Transformations` seems good

File: docs/source/reference/image_classification.rst (hunk 2 above)
Comment (aniketmaurya) [reply docs/source/reference/image_classification.rst:66]: didn't get what you mean by "Let's use testcode here" 😅. Could you pls explain this?

**Pull request merged**